BookingV2.Logic/Services/RealEstateService.cs

```diff
@@ -0,0 +1,167 @@
+using BookingV2.Logic.Contract;
+using BookingV2.Logic.Responses;
+using BookinV2.Data;
+using BookinV2.Data.Entities.RealEstateEntities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace BookingV2.Logic.Services
+{
+    public class RealEstateService : IRealEstateService
+    {
+        private readonly BookingV2DBContext _context;
+        private readonly ILogger<RealEstateService> _logger;
+
+        public RealEstateService(BookingV2DBContext context, ILogger<RealEstateService> logger)
+        {
+            this._context = context ?? throw new ArgumentNullException(nameof(context));
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<ApiResponse<IEnumerable<RealEstate>>> GetAllAsync()
+        {
+            var response = new ApiResponse<IEnumerable<RealEstate>>();
+
+            try
+            {
+                if (this._context.RealEstates == null)
+                {
+                    response.AddError("RealEstate DbSet is null.");
+                    return response;
+                }
+
+                response.Response = await this._context.RealEstates.ToListAsync();
+            }
+            catch (Exception ex)
+            {
+                this._logger.LogError($"Failed to get all RealEstates: {ex.Message}");
+                response.AddError("An error occurred while fetching data.");
+            }
+
+            return response;
+        }
+
+        public async Task<ApiResponse<RealEstate>> GetByIdAsync(int id)
+        {
+            var response = new ApiResponse<RealEstate>();
+
+            try
+            {
+                if (this._context.RealEstates == null)
+                {
+                    response.AddError("RealEstate DbSet is null.");
+                    return response;
+                }
+
+                var realEstate = await this._context.RealEstates.FindAsync(id);
+
+                if (realEstate == null)
+                {
+                    response.AddError("RealEstate not found.");
+                    return response;
+                }
+
+                response.Response = realEstate;
+            }
+            catch (Exception ex)
+            {
+                this._logger.LogError($"Failed to get RealEstate by id {id}: {ex.Message}");
+                response.AddError("An error occurred while fetching data.");
+            }
+
+            return response;
+        }
+
+        public async Task<ApiResponse<RealEstate>> AddAsync(RealEstate realEstate)
```

Comment:
services should use their level(Service level BLL) models in order to not mix app logic levels. So please create new type for service model for input